# Ultralytics Framework for YOLOv10 Training

In [1]:
import torch

# Number of GPUs available
num_gpus = torch.cuda.device_count()

print(f"Number of available GPUs: {num_gpus}")


Number of available GPUs: 4


# Yolov10s Model Training on:
- Carla_Labelling-v5 dataset
- Train: 70% = 4485 Images
- Validation: 20% = 1274 Images
- Test: 10% = 641 Images

- Hyperparameters are on automatic settings using Ultralytics package

In [2]:
from ultralytics import YOLO

# Load YOLOv10n model from scratch
model = YOLO("yolov10s.pt")

100%|██████████| 15.9M/15.9M [00:00<00:00, 105MB/s] 


# Validate the pretrained weights by running inference on an image.

In [3]:
# Run batched inference on a list of images
results = model(["/home/sur06423/project/1.jpg"])  # return a list of Results objects

# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk


0: 448x640 4 horses, 63.1ms
Speed: 6.3ms preprocess, 63.1ms inference, 24.3ms postprocess per image at shape (1, 3, 448, 640)


Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)


# Train the model using the following guidelines:
- Link for various arguments: https://docs.ultralytics.com/modes/train/#train-settings 

In [4]:
import os

# Set library paths
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:' + os.environ.get('LD_LIBRARY_PATH', '')
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']

# Verify the update
print(os.environ['LD_LIBRARY_PATH'])


/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:/usr/lib/xorg/lib/x86_64-linux-gnu:/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:/home/sur06423/miniconda3/envs/yolov_env/lib/python3.9/site-packages/cv2/../../lib64:


In [5]:
# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
model.train(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", # 	Path to the dataset configuration file (e.g., coco8.yaml). This file contains dataset-specific parameters, including paths to training and validation data, class names, and number of classes.
            epochs=100, # Total number of training epochs. Each epoch represents a full pass over the entire dataset. Adjusting this value can affect training duration and model performance.
            patience=10, # Number of epochs to wait without improvement in validation metrics before early stopping the training. Helps prevent overfitting by stopping training when performance plateaus.
            batch=128, # Batch size, with three modes: set as an integer (e.g., batch=16), auto mode for 60% GPU memory utilization (batch=-1), or auto mode with specified utilization fraction (batch=0.70).
            imgsz=640, # Target image size for training. All images are resized to this dimension before being fed into the model. Affects model accuracy and computational complexity.
            save=True, # Enables saving of training checkpoints and final model weights. Useful for resuming training or model deployment.
            save_period = 1, # Frequency of saving model checkpoints, specified in epochs. A value of -1 disables this feature. Useful for saving interim models during long training sessions.
            device=[0,1,2,3], # Specifies the computational device(s) for training: a single GPU (device=0), multiple GPUs (device=0,1), CPU (device=cpu), or MPS for Apple silicon (device=mps).
            workers=6, # Number of worker threads for data loading (per RANK if Multi-GPU training). Influences the speed of data preprocessing and feeding into the model, especially useful in multi-GPU setups.
            project="/home/sur06423/project/Version_5_exp/YOLO_v10/v10_s", # Name of the project directory where training outputs are saved. Allows for organized storage of different experiments.
            pretrained = True, # Determines whether to start training from a pretrained model. Can be a boolean value or a string path to a specific model from which to load weights. Enhances training efficiency and model performance.
            optimizer = "auto", # Choice of optimizer for training. Options include SGD, Adam, AdamW, NAdam, RAdam, RMSProp etc., or auto for automatic selection based on model configuration. Affects convergence speed and stability.
            verbose = True, # Enables verbose output during training, providing detailed logs and progress updates. Useful for debugging and closely monitoring the training process.
            seed = 0, # Sets the random seed for training, ensuring reproducibility of results across runs with the same configurations.
            amp = True, # Enables Automatic Mixed Precision (AMP) training, reducing memory usage and possibly speeding up training with minimal impact on accuracy.
            profile =True, # Enables profiling of ONNX and TensorRT speeds during training, useful for optimizing model deployment.
            val=True, # Enables validation during training, allowing for periodic evaluation of model performance on a separate dataset.
            plots = True, #Generates and saves plots of training and validation metrics, as well as prediction examples, providing visual insights into model performance and learning progression.                
)

New https://pypi.org/project/ultralytics/8.3.13 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
engine/trainer: task=detect, mode=train, model=yolov10s.pt, data=/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml, epochs=100, time=None, patience=10, batch=128, imgsz=640, save=True, save_period=1, cache=False, device=[0, 1, 2, 3], workers=6, project=/home/sur06423/project/Version_5_exp/YOLO_v10/v10_s, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0

Overriding model.yaml nc=80 with nc=4

                   from  n    params  module                                       arguments                     
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]                 
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  2                  -1  1     29056  ultralytics.nn.modules.block.C2f             [64, 64, 1, True]             
  3                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  4                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  5                  -1  1     36096  ultralytics.nn.modules.block.SCDown          [128, 256, 3, 2]              
  6                  -1  2    788480  ultralytics.nn.modules.block.C2f             [256, 256, 2, True]           
  7                  -1  1    137728  ultralytics

100%|██████████| 6.25M/6.25M [00:00<00:00, 99.1MB/s]


AMP: checks passed ✅


train: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/valid/labels.cache... 1274 images, 124 backgrounds, 0 corrupt: 100%|██████████| 1274/1274 [00:00<?, ?it/s]


Plotting labels to /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 99 weight(decay=0.0), 112 weight(decay=0.001), 111 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 24 dataloader workers
Logging results to /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      10.1G      2.704       16.5      2.131          5        640: 100%|██████████| 36/36 [00:14<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.64it/s]


                   all       1274       7377      0.358      0.224      0.218      0.152

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      10.2G      2.479      2.712      1.975         17        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.92it/s]


                   all       1274       7377      0.545      0.377      0.293      0.161

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      10.1G      2.535      2.085       2.01         37        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.97it/s]


                   all       1274       7377      0.406      0.343      0.239      0.114

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      10.1G      2.492      1.945      2.021          8        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.94it/s]


                   all       1274       7377      0.408      0.377      0.288      0.154

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      10.1G       2.47      1.809      1.973          6        640: 100%|██████████| 36/36 [00:11<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.98it/s]


                   all       1274       7377      0.604       0.41      0.406      0.229

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      10.1G      2.374      1.653      1.932         26        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.93it/s]


                   all       1274       7377      0.613      0.446      0.452      0.264

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      10.2G       2.37      1.639      1.941          3        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.04it/s]


                   all       1274       7377      0.751      0.438      0.501      0.285

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      10.1G      2.327      1.556      1.892         24        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.674      0.458      0.495      0.287

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100      10.1G      2.237      1.493      1.907          1        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377       0.77      0.494      0.572      0.336

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100        10G      2.245      1.418      1.874          8        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.04it/s]


                   all       1274       7377      0.778       0.48       0.55       0.33

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100        10G      2.305       1.44      1.867         14        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.93it/s]


                   all       1274       7377       0.74      0.512      0.561      0.345

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100      10.1G       2.23      1.492       1.83          1        640: 100%|██████████| 36/36 [00:11<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.63it/s]


                   all       1274       7377      0.782      0.518      0.587       0.35

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      10.2G      2.161      1.337      1.855         11        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.88it/s]


                   all       1274       7377       0.76      0.527      0.586       0.36

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      10.1G      2.201      1.437      1.846         14        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.786      0.537      0.601      0.376

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      10.1G      2.152      1.304      1.837         24        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.789      0.546      0.598      0.356

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      10.1G       2.13      1.301      1.835          4        640: 100%|██████████| 36/36 [00:11<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  4.00it/s]


                   all       1274       7377      0.788      0.558      0.618      0.382

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      10.2G      2.102       1.28      1.838         15        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.97it/s]


                   all       1274       7377      0.775       0.55      0.612      0.377

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100        10G      2.081      1.313      1.851          6        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.97it/s]


                   all       1274       7377       0.81      0.525      0.614      0.383

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      10.4G      2.159      1.274      1.836         12        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.89it/s]


                   all       1274       7377      0.788       0.56      0.624      0.385

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      10.2G      2.078       1.24      1.827         23        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377       0.79      0.543      0.614      0.374

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      10.2G      2.048      1.198      1.815          4        640: 100%|██████████| 36/36 [00:11<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.91it/s]


                   all       1274       7377      0.818       0.55      0.627      0.389

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100      10.1G      2.089      1.241      1.817          3        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.98it/s]


                   all       1274       7377      0.816      0.563      0.639      0.405

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      10.2G      2.034      1.201      1.813         27        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.98it/s]


                   all       1274       7377      0.807      0.566      0.633      0.391

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      10.1G      2.012      1.161      1.808          3        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.96it/s]


                   all       1274       7377       0.83      0.566      0.645      0.407

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100      10.1G      1.998      1.172      1.801         20        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.96it/s]


                   all       1274       7377      0.826      0.578      0.648      0.407

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      10.2G      1.974      1.151      1.823         18        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.826      0.575      0.652      0.413

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      10.1G      2.027      1.119      1.797         11        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.816      0.585      0.657       0.41

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100        10G      1.951      1.123      1.795         20        640: 100%|██████████| 36/36 [00:11<00:00,  3.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.98it/s]


                   all       1274       7377      0.833      0.591       0.65      0.407

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100      10.1G      1.937      1.097      1.782          7        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.95it/s]


                   all       1274       7377       0.84      0.573       0.65      0.412

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      10.1G      1.974      1.115       1.81         11        640: 100%|██████████| 36/36 [00:11<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.65it/s]


                   all       1274       7377      0.822       0.58      0.648      0.411

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      10.1G      1.942      1.099      1.793         19        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.849      0.582      0.663       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      10.1G      1.937      1.074      1.776         10        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  4.00it/s]


                   all       1274       7377      0.826      0.585      0.656      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      10.1G      1.888      1.061      1.774         13        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  4.00it/s]


                   all       1274       7377      0.847      0.583      0.663       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      10.1G      1.912      1.054       1.79         11        640: 100%|██████████| 36/36 [00:11<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.00it/s]


                   all       1274       7377       0.85      0.578      0.656      0.416

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100      10.2G      1.935       1.09        1.8         21        640: 100%|██████████| 36/36 [00:11<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.818      0.601      0.674       0.43

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      10.1G      1.895      1.047       1.78          9        640: 100%|██████████| 36/36 [00:12<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.43it/s]


                   all       1274       7377      0.858      0.586      0.659      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      10.1G      1.906      1.055      1.789         10        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.95it/s]


                   all       1274       7377      0.842      0.593      0.672      0.427

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      10.1G      1.808      1.043      1.737          0        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.64it/s]


                   all       1274       7377      0.829      0.599      0.674      0.428

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      10.1G      1.878      1.027      1.773         21        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.93it/s]


                   all       1274       7377      0.838      0.591      0.662      0.428

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100        10G      1.867      1.031      1.769         15        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.97it/s]


                   all       1274       7377      0.821      0.601      0.668      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      10.1G      1.843      1.031      1.774         10        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.823      0.606      0.677       0.43

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100      10.1G      1.864      1.032      1.769         24        640: 100%|██████████| 36/36 [00:11<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  4.00it/s]


                   all       1274       7377       0.83      0.613      0.683      0.439

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      10.1G      1.806      1.009      1.777          4        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.97it/s]


                   all       1274       7377      0.849      0.602      0.676      0.439

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      10.4G      1.871      1.005      1.747         26        640: 100%|██████████| 36/36 [00:11<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.95it/s]


                   all       1274       7377      0.862      0.612      0.691      0.444

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100      10.1G      1.789     0.9632      1.762          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.00it/s]


                   all       1274       7377      0.847      0.614      0.687      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      10.1G      1.768     0.9588      1.758          8        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.836      0.614      0.688      0.443

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      10.1G      1.807     0.9659      1.753         15        640: 100%|██████████| 36/36 [00:11<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.843        0.6      0.679      0.439

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      10.1G      1.841     0.9707      1.753         16        640: 100%|██████████| 36/36 [00:11<00:00,  3.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.61it/s]


                   all       1274       7377      0.821      0.623      0.688      0.442

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100        10G      1.826       1.08      1.725          7        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.97it/s]


                   all       1274       7377       0.86      0.603      0.687      0.444

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100        10G      1.858     0.9876      1.755         17        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.843      0.623      0.691      0.448

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      10.2G      1.789     0.9468      1.766         10        640: 100%|██████████| 36/36 [00:11<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.93it/s]


                   all       1274       7377      0.844      0.612      0.688      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100      10.1G      1.763     0.9385      1.757          5        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.844      0.621      0.698      0.453

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100        10G      1.757     0.9355      1.762          8        640: 100%|██████████| 36/36 [00:11<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.847       0.62      0.705      0.457

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100        10G      1.707     0.9177      1.738         14        640: 100%|██████████| 36/36 [00:11<00:00,  3.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.99it/s]


                   all       1274       7377      0.836      0.619      0.693      0.447

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      10.1G      1.779     0.9586      1.746         21        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.853      0.617      0.696      0.452

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100      10.1G      1.704     0.9105      1.746          5        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.825      0.621      0.693      0.448

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      10.2G      1.763     0.9202      1.738         13        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.06it/s]


                   all       1274       7377      0.834      0.626      0.701      0.454

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100      10.1G       1.73     0.9189      1.744         21        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.98it/s]


                   all       1274       7377      0.867      0.613      0.703       0.46

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      10.1G      1.734     0.9032      1.736         12        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.99it/s]


                   all       1274       7377      0.843      0.636      0.709      0.463

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      10.1G      1.736     0.9155      1.739         18        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.04it/s]


                   all       1274       7377       0.85       0.63      0.704      0.459

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100      10.1G       1.75     0.9063      1.732         39        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.00it/s]


                   all       1274       7377      0.846      0.634      0.706      0.459

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      10.1G      1.678     0.8758      1.734         33        640: 100%|██████████| 36/36 [00:11<00:00,  3.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377       0.84      0.633      0.703      0.463

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100      10.1G       1.74     0.8991      1.731         13        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.865      0.623      0.713      0.469

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100      10.1G       1.75     0.8917      1.723          6        640: 100%|██████████| 36/36 [00:11<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.854      0.623      0.708      0.467

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      10.1G      1.757     0.9168      1.738          3        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.04it/s]


                   all       1274       7377      0.867      0.626       0.71       0.47

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      10.1G      1.705     0.8919      1.736         10        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.07it/s]


                   all       1274       7377      0.889      0.635      0.724      0.472

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      10.1G      1.743     0.8991       1.74         19        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.863      0.633      0.711      0.467

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100      10.1G       1.69     0.8614      1.712         19        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  4.00it/s]


                   all       1274       7377       0.84      0.638      0.709      0.466

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      10.2G      1.718     0.8885      1.738         16        640: 100%|██████████| 36/36 [00:11<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.80it/s]


                   all       1274       7377      0.848      0.637      0.713      0.469

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      10.1G      1.702     0.8658      1.726         24        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.85it/s]


                   all       1274       7377      0.845      0.641      0.715      0.475

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      10.1G      1.675     0.8466      1.727         17        640: 100%|██████████| 36/36 [00:11<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.863      0.629      0.711      0.469

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      10.1G      1.678     0.8485      1.724         12        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.852      0.636      0.717      0.472

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      10.1G      1.702     0.8632      1.735         13        640: 100%|██████████| 36/36 [00:11<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  4.00it/s]


                   all       1274       7377      0.876       0.63      0.718      0.472

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      10.2G      1.632     0.8398      1.732         11        640: 100%|██████████| 36/36 [00:11<00:00,  3.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.865      0.632      0.722      0.478

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      10.1G      1.678     0.8447       1.73          6        640: 100%|██████████| 36/36 [00:11<00:00,  3.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.99it/s]


                   all       1274       7377      0.866      0.639      0.724      0.475

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      10.1G      1.668     0.8282      1.714          8        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.855      0.638       0.72      0.477

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      10.1G      1.685     0.8532      1.722         22        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.62it/s]


                   all       1274       7377      0.855       0.63      0.718      0.474

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100      10.1G       1.64     0.8301      1.736          4        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.839      0.648       0.72      0.475

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      10.1G      1.639     0.8156      1.724         13        640: 100%|██████████| 36/36 [00:11<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.92it/s]


                   all       1274       7377      0.861      0.639       0.72      0.478

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      10.1G      1.625      0.825      1.719         36        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.865      0.642      0.724      0.477

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100      10.1G      1.618     0.8401      1.715          7        640: 100%|██████████| 36/36 [00:11<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.05it/s]


                   all       1274       7377      0.867      0.641      0.725       0.48

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      10.2G      1.614     0.8275      1.715          9        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.872       0.64      0.726      0.482

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      10.1G      1.618     0.8123      1.717          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.99it/s]


                   all       1274       7377      0.889      0.633      0.725      0.483

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      10.1G      1.641     0.8391      1.717          8        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.871      0.639      0.728      0.485

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100      10.3G      1.638     0.8161      1.726         11        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.97it/s]


                   all       1274       7377      0.878      0.642      0.731      0.488

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100      10.1G       1.71      1.032      1.901          3        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.96it/s]


                   all       1274       7377      0.864      0.645      0.723       0.48

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100        10G      1.626     0.8067       1.72         14        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.48it/s]


                   all       1274       7377      0.866      0.639       0.73      0.486

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100      10.1G      1.608     0.7896      1.711         16        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.94it/s]


                   all       1274       7377      0.855      0.654      0.732      0.485

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      10.1G      1.523     0.7373      1.718         14        640: 100%|██████████| 36/36 [00:11<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.98it/s]


                   all       1274       7377      0.866      0.646       0.73      0.486

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      10.1G      1.568     0.7783      1.731          3        640: 100%|██████████| 36/36 [00:11<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.00it/s]


                   all       1274       7377      0.865      0.644      0.726      0.485
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      10.1G      1.691     0.7951       1.69         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.865      0.647      0.728      0.485

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100      10.1G      1.707     0.7981      1.695         13        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.867      0.643      0.728      0.486

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      10.1G      1.675     0.7847      1.684          7        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.05it/s]


                   all       1274       7377       0.86       0.65      0.729      0.488

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      10.1G      1.705      0.804      1.691          6        640: 100%|██████████| 36/36 [00:11<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.03it/s]


                   all       1274       7377      0.873      0.647      0.729      0.489

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      10.2G      1.724     0.8161      1.683         31        640: 100%|██████████| 36/36 [00:11<00:00,  3.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.98it/s]


                   all       1274       7377      0.872      0.648       0.73      0.487

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      10.1G      1.688       0.81      1.693         13        640: 100%|██████████| 36/36 [00:11<00:00,  3.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.78it/s]


                   all       1274       7377      0.873      0.645      0.732      0.489

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      10.1G      1.688     0.7899      1.686         13        640: 100%|██████████| 36/36 [00:11<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.04it/s]


                   all       1274       7377      0.881       0.64      0.732      0.493

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      10.1G      1.675     0.7794      1.673          3        640: 100%|██████████| 36/36 [00:11<00:00,  3.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377      0.867      0.649      0.735      0.492

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      10.1G      1.667     0.7775      1.681         40        640: 100%|██████████| 36/36 [00:11<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  4.00it/s]


                   all       1274       7377      0.867      0.649      0.737      0.494

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100      10.1G      1.645     0.7571      1.681          6        640: 100%|██████████| 36/36 [00:11<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.04it/s]


                   all       1274       7377      0.874      0.645      0.734      0.492

100 epochs completed in 0.575 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/last.pt, 16.5MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/best.pt, 16.5MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:08<00:00,  2.40it/s]


                   all       1274       7377      0.867       0.65      0.737      0.494
            Pedestrian        333        600      0.789      0.673      0.737      0.439
         Traffic_Signs        309        434      0.894      0.585      0.693      0.416
               Vehicle       1034       2820      0.923      0.888      0.939      0.758
         traffic_light        660       3523      0.862      0.452      0.578      0.362
Speed: 0.1ms preprocess, 1.9ms inference, 0.0ms loss, 1.5ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train


# Results on Validation set:
100 epochs completed in 0.575 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/last.pt, 16.5MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/best.pt, 16.5MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:08<00:00,  2.40it/s]
                   all       1274       7377      0.867       0.65      0.737      0.494
            Pedestrian        333        600      0.789      0.673      0.737      0.439
         Traffic_Signs        309        434      0.894      0.585      0.693      0.416
               Vehicle       1034       2820      0.923      0.888      0.939      0.758
         traffic_light        660       3523      0.862      0.452      0.578      0.362
Speed: 0.1ms preprocess, 1.9ms inference, 0.0ms loss, 1.5ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train

# Test the results on Test dataset
- The above results was on the validation set of the Carla_labeling-v5 dataset (See the Image counts)
- Now we need to calculate the metrics on the test set of the Carla_labeling-v5 dataset containing only 641 images.
- Below is the script to do so:

In [6]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.69s/it]


                   all        641       3701      0.837      0.625      0.765      0.559
            Pedestrian        182        332      0.745      0.651      0.753      0.494
         Traffic_Signs        168        230      0.831      0.535      0.715      0.498
               Vehicle        528       1377      0.915      0.879       0.93      0.799
         traffic_light        338       1762      0.858      0.435      0.663      0.446
Speed: 0.3ms preprocess, 3.8ms inference, 0.0ms loss, 0.1ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val


# Results on Last checpoint (100th epoch):
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.69s/it]
                   all        641       3701      0.837      0.625      0.765      0.559
            Pedestrian        182        332      0.745      0.651      0.753      0.494
         Traffic_Signs        168        230      0.831      0.535      0.715      0.498
               Vehicle        528       1377      0.915      0.879       0.93      0.799
         traffic_light        338       1762      0.858      0.435      0.663      0.446
Speed: 0.3ms preprocess, 3.8ms inference, 0.0ms loss, 0.1ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val

# Results on Test Set using Best Checkpoint:

In [7]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load the model
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.67s/it]


                   all        641       3701      0.867       0.61      0.765       0.56
            Pedestrian        182        332      0.799      0.636      0.755      0.495
         Traffic_Signs        168        230      0.842      0.504      0.708      0.497
               Vehicle        528       1377      0.933      0.875      0.932        0.8
         traffic_light        338       1762      0.893      0.425      0.666      0.447
Speed: 0.5ms preprocess, 3.6ms inference, 0.0ms loss, 0.1ms postprocess per image
Saving runs/detect/val2/predictions.json...
Results saved to runs/detect/val2


In [15]:
print(checkpoint.keys())

dict_keys(['date', 'version', 'license', 'docs', 'epoch', 'best_fitness', 'model', 'ema', 'updates', 'optimizer', 'train_args', 'train_metrics', 'train_results'])


# Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv10s - 0.233
- IoU threshold : 0.5
- Results save d at: runs/detect/val3

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.233, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.32s/it]


                   all        641       3701      0.829      0.627      0.764      0.558
            Pedestrian        182        332      0.732      0.651       0.75      0.492
         Traffic_Signs        168        230      0.826      0.535      0.713      0.497
               Vehicle        528       1377      0.909      0.882      0.931      0.799
         traffic_light        338       1762      0.849       0.44      0.663      0.445
Speed: 1.9ms preprocess, 2.9ms inference, 0.0ms loss, 0.3ms postprocess per image
Saving runs/detect/val3/predictions.json...
Results saved to runs/detect/val3


# Best Checkpoint + Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv10s - 0.233
- IoU threshold : 0.5
- Results save d at: runs/detect/val4

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.233, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:06<00:00,  1.12s/it]


                   all        641       3701      0.857      0.613      0.765      0.558
            Pedestrian        182        332      0.784      0.639      0.755      0.494
         Traffic_Signs        168        230       0.83      0.509      0.706      0.494
               Vehicle        528       1377      0.933      0.877      0.933      0.799
         traffic_light        338       1762      0.883      0.427      0.665      0.446
Speed: 1.7ms preprocess, 2.8ms inference, 0.0ms loss, 0.3ms postprocess per image
Saving runs/detect/val4/predictions.json...
Results saved to runs/detect/val4


# Best Checkpoint + Optimal Confidence + Best Epoch 99 + IoU = 0.6
- Optimal Confidence YOLOv10s - 0.233
- IoU threshold : 0.6
- Results save d at: runs/detect/val5

In [2]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.233, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.24s/it]


                   all        641       3701      0.857      0.613      0.765      0.558
                   all        641       3701      0.857      0.613      0.765      0.558
            Pedestrian        182        332      0.784      0.639      0.755      0.494
            Pedestrian        182        332      0.784      0.639      0.755      0.494
         Traffic_Signs        168        230       0.83      0.509      0.706      0.494
         Traffic_Signs        168        230       0.83      0.509      0.706      0.494
               Vehicle        528       1377      0.933      0.877      0.933      0.799
               Vehicle        528       1377      0.933      0.877      0.933      0.799
         traffic_light        338       1762      0.883      0.427      0.665      0.446
         traffic_light        338       1762      0.883      0.427      0.665      0.446
Speed: 2.0ms preprocess, 2.7ms inference, 0.0ms loss, 0.3ms postprocess per image
Speed: 2.0ms preprocess, 2.7

# Seperate evaluation on the training set

In [1]:
from ultralytics import YOLO

# Load your model
model = YOLO('/home/sur06423/project/Version_5_exp/YOLO_v10/v10_s/train/weights/best.pt')  # replace with your model path

# Validate on custom dataset
results = model.val(data='/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml',
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the optimal confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="train", 
)  # Ensure your YAML points to your desired dataset

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10s summary (fused): 293 layers, 8,038,056 parameters, 0 gradients, 24.5 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 36/36 [00:26<00:00,  1.36it/s]


                   all       4485      26788      0.885      0.672      0.808      0.618
            Pedestrian       1229       2227      0.832      0.721      0.823      0.594
         Traffic_Signs       1093       1459      0.884      0.629      0.784       0.55
               Vehicle       3636      10031      0.931      0.906      0.946      0.833
         traffic_light       2373      13071      0.891      0.434      0.678      0.493
Speed: 0.4ms preprocess, 2.5ms inference, 0.0ms loss, 0.1ms postprocess per image
Saving runs/detect/val6/predictions.json...
Results saved to runs/detect/val6


# This notebook is complete.
- All the cell are executed and saved for later use.
- Results are saved in the corresponding directories.